src/sdam/monitor.ts

```diff
@@ -44,6 +45,11 @@ function isInCloseState(monitor: Monitor) {
   return monitor.s.state === STATE_CLOSED || monitor.s.state === STATE_CLOSING;
 }
 
+/** @public */
+export const ServerMonitoringModes = ['auto', 'poll', 'stream'];
```

Comment:
If this is to be a public API then wouldn't an enum be more ergonomic? Then we can use `Object.values()` to get the `.includes` in the options parsing working.

Reply:
I've updated to be an enum now and updated the options parsing.